test/e2e/specs/editor/various/rich-text.spec.js

```diff
@@ -0,0 +1,826 @@
+			{
+				name: 'core/paragraph',
+				attributes: { content: '<strong><em>1</em></strong>.' },
+			},
+		] );
+	} );
+
+	test( 'should not highlight more than one format', async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await pageUtils.pressKeys( 'primary+b' );
+		await page.keyboard.type( '1' );
+		await pageUtils.pressKeys( 'primary+b' );
+		await page.keyboard.type( ' 2' );
+		await pageUtils.pressKeys( 'shift+ArrowLeft' );
+		await pageUtils.pressKeys( 'primary+b' );
+
+		const count = await editor.canvas.evaluate(
+			() =>
+				document.querySelectorAll( '*[data-rich-text-format-boundary]' )
+					.length
+		);
+		expect( count ).toBe( 1 );
+	} );
+
+	test( 'should return focus when pressing formatting button', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'Some ' );
+		await editor.clickBlockToolbarButton( 'Bold' );
+		await page.keyboard.type( 'bold' );
+		await editor.clickBlockToolbarButton( 'Bold' );
+		await page.keyboard.type( '.' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: { content: 'Some <strong>bold</strong>.' },
+			},
+		] );
+	} );
+
+	test( 'should transform backtick to code', async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'A `backtick`' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: { content: 'A <code>backtick</code>' },
+			},
+		] );
+
+		await pageUtils.pressKeys( 'primary+z' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [
+			{ name: 'core/paragraph' },
+		] );
+	} );
+
+	test( 'should undo backtick transform with backspace', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '`a`' );
+		await page.keyboard.press( 'Backspace' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: { content: '`a`' },
+			},
+		] );
+	} );
+
+	test( 'should not undo backtick transform with backspace after typing', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '`a`' );
+		await page.keyboard.type( 'b' );
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.press( 'Backspace' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [] );
+	} );
+
+	test( 'should not undo backtick transform with backspace after selection change', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '`a`' );
+		await page.evaluate( () => new Promise( window.requestIdleCallback ) );
```

Comment:
I've said this before elsewhere but repeating it in case you missed it:  I don't think e2e tests should rely on implementation details like this. I understand if this is just a 1-to-1 migration though! Flagging this again so that we can find another way to solve this in the future. c.c. @Mamaduka any idea?

Reply:
I welcome solutions. We have to specifically wait this amount of time, otherwise the test could be running too fast and we'd get flaky tests. There's code that's waiting for an idle callback.

Reply:
I think we can make this test a bit more explicit in terms of what's actually happening on the screen, which might allow us to drop the `requestIdleCallback`. For example:

```js
test( 'should not undo backtick transform with backspace after selection change', async ( {
	page,
	pageUtils,
	editor,
} ) => {
	await editor.canvas
		.getByRole( 'button', { name: 'Add default block' } )
		.click();
	await page.keyboard.type( '`test`' );

	const inlineCode = editor.canvas
		.getByRole( 'document', {
			name: 'Paragraph block',
		} )
		.locator( 'code' );

	await expect( inlineCode ).toHaveText( 'test' ); // Here we make sure the inline code exists before moving inside the format boundary.

	// Move inside format boundary.
	await pageUtils.pressKeys( 'ArrowLeft', { times: 2 } );
	await pageUtils.pressKeys( 'ArrowRight' );

	// Empty the inline code content.
	await pageUtils.pressKeys( 'Backspace', { times: 4 } );
	await expect( inlineCode ).not.toBeAttached();

	await page.keyboard.type( 'test' );
	await expect( inlineCode ).toHaveText( 'test' );
} );

```